spec/classes/profile/solr_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+require 'spec_helper'
+
+describe 'nebula::profile::solr' do
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:facts) { os_facts }
+
+      it { is_expected.to compile }
+
+      context 'with package' do
+        [
+          'openjdk-8-jre-headless',
+          'solr',
+          'lsof',
+        ].each do |package|
+          it { is_expected.to contain_package(package) }
+        end
+      end
+
+      context 'with directory' do
+        it do
+          [
+            '/var/lib/solr',
+            '/var/lib/solr/home',
+            '/var/lib/solr/logs',
+          ].each do |path|
+            is_expected.to contain_file(path).with(
```

Comment:
This should be inverted. Each of these should get its own `it do`. This is so they show as itemized examples and anything that fails will do so individually.

Reply:
Same applies below.

Reply:
Okay fixed them.

One final blessing and I will merge it on Monday (to avoid Friday Fiascos).